app/src/test/adminUsers.test.tsx

```diff
@@ -1,12 +1,421 @@
-import React from 'react';
-import { render } from '@testing-library/react';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import { BrowserRouter } from 'react-router-dom';
+import '@testing-library/jest-dom';
 import Users from '../admin/users';
+import adminApi from '../admin/adminApi';
+import { act } from 'react-dom/test-utils';
 
+jest.mock('../admin/adminApi');
+jest.mock('../contexts/AuthContext', () => ({
+  ...jest.requireActual('../contexts/AuthContext'), // use actual for all non-hook parts
+  useAuth: () => ({
+    currentUser: {
+      id: 'mocked-user-id',
+      role: 'admin',
+      email: 'mocked-email@ualberta.ca'
+    },
+    sessionToken: 'mocked-session-token',
+    login: jest.fn(),
+    logout: jest.fn(),
+  })
+}));
 
-//jest.useFakeTimers({ legacyFakeTimers: true });
-beforeAll(() => {
-    jest.useFakeTimers('legacy');
+//Mocking useAuth
+const useAuthMock = jest.spyOn(require('../contexts/AuthContext'), 'useAuth');
+
+//Mock responses for each of the api calls made in admin users page
+const mockDishesStatusData = [
+  {
+    userId: 'user1',
+    emailAddress: 'hello@ualberta.ca',
+    inUse: 2,
+    overdue: 30,
+    role: 'customer',
+  },
+  {
+    userId: 'user2',
+    emailAddress: 'user2@ualberta.ca',
+    inUse: 2,
+    overdue: 5,
+    role: 'admin',
+  },
+  {
+    userId: 'user3',
+    emailAddress: 'mysterious@ualberta.ca',
+    inUse: 3,
+    overdue: 39,
+    role: 'volunteer',
+  },
+  {
+    userId: 'user4',
+    emailAddress: 'mocked-email@ualberta.ca',
+    inUse: 5,
+    overdue: 45,
+    role: 'admin',
+  }
+];
+
+const mockInUseResponse = [
+  {
+    email: 'hello@ualberta.ca',
+    count: 2
+
+  },
+  {
+    email: 'user2@ualberta.ca',
+    count: 2
+
+  },
+  {
+    email: 'mysterious@ualberta.ca',
+    count: 3
+
+  },
+  {
+    email: 'mocked-email@ualberta.ca',
+    count: 5
+  }
+  
+];
+
+const mockOverdueResponse = [
+  {
+    email: 'hello@ualberta.ca',
+    count: 30
+
+  },
+  {
+    email: 'user2@ualberta.ca',
+    count: 5
+
+  },
+  {
+    email: 'mysterious@ualberta.ca',
+    count: 39
+
+  },
+  {
+    email: 'mocked-email@ualberta.ca',
+    count: 45
+  }
+]
+
+const mockDishes = [
+  {
+    id: 1,
+    qid: 2,
+    registered: '2023-07-22T20:19:47.144Z',
+    status: 'available',
+    timesBorrowed: 7,
+    type: "plate",
+    userId: "yo"
+
+  }
+]
+
+const mockUsers = [
+  {
+    userId: 'user1',
+    emailAddress: 'hello@ualberta.ca',
+    role: 'customer',
+  },
+  {
+    userId: 'user2',
+    emailAddress: 'user2@ualberta.ca',
+    role: 'admin',
+  },
+  {
+    userId: 'user3',
+    emailAddress: 'mysterious@ualberta.ca',
+    role: 'volunteer',
+  },
+  {
+    userId: 'user4',
+    emailAddress: 'mocked-email@ualberta.ca',
+    role: 'admin'
+  }
+]
+
+const mockModifyRole = {
+  data:{
+  status: "Success"
+  }
+}
+
+
+beforeEach(async () => {
+    jest.clearAllMocks(); // Clears any previous mocking behavior before each test
```

Comment:
a little heavy handed with the comments here, try to only use comments to explain something that is very complicated or your reason for doing something 😄 

```suggestion
    jest.clearAllMocks(); 
```